### PR TITLE
feat: support env context in `dev:exec` and `build`

### PR DESCRIPTION
**Pull request by jasonbarry**
#### Summary

Fixes https://github.com/netlify/pillar-workflow/issues/803

This PR supports the `--context` flag in `netlify dev:exec` and `netlify build` for pulling environment variables from the correct context in Envelope. 

`--context` was already allowed in `netlify build` (same with reading from `process.env.CONTEXT`), but it only read env vars and context from the toml file. Now that context-aware env vars are supported in the UI for sites opted in to Envelope, we need to make a network request to check if there are any env vars that have been defined in the UI. Values in the toml file will still be respected too. 

It also now respects the previously defined `--offline` flag, so no calls will go out to Envelope if that's present. 

---

For us to review and ship your PR efficiently, please perform the following steps:

- [x] Open a [bug/issue](https://github.com/netlify/cli/issues/new/choose) before writing your code 🧑‍💻. This ensures we can discuss the changes and get feedback from everyone that should be involved. If you\`re fixing a typo or something that\`s on fire 🔥 (e.g. incident related), you can skip this step.
- [x] Read the [contribution guidelines](../CONTRIBUTING.md) 📖. This ensures your code follows our style guide and
      passes our tests.
- [ ] Update or add tests (if any source code was changed or added) 🧪
- [x] Update or add documentation (if features were changed or added) 📝
- [x] Make sure the status checks below are successful ✅

**A picture of a cute animal (not mandatory, but encouraged)**

Screenshot 2022-08-15 at 13 04 27

### CODE REVIEW

**Review by melanierichards — COMMENTED** [review]
Thanks JB! Opened up one thread where I'm thinking aloud. :)

File: src/commands/build/build.js [hunk 1]
```diff
@@ -77,8 +85,8 @@ const createBuildCommand = (program) =>
   program
     .command('build')
     .description('(Beta) Build on your local machine')
+    .option('--context <context>', 'Specify a build context', process.env.CONTEXT || 'production')
```
[the remark this answers is not in the document]
Comment (jasonbarry) [reply src/commands/build/build.js:88]: @melanierichards it is indeed intentional! And is the existing behavior per this test https://github.com/netlify/cli/blob/main/tests/integration/110.command.build.test.js#L92

**Comment by jasonbarry** [comment]
@sgrove I moved around some logic related to `authlifyTokenId` -- would you mind taking a look to see that nothing is negatively impacted?

File: src/commands/build/build.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (codebyuma) [reply src/commands/build/build.js:88]: Just to confirm, @melanierichards do we want to add this `netlify build` update to the Labs docs? Right now, we don't mention the command but we could mention that it'll also read values set in the Netlify UI now.

**Review by codebyuma — COMMENTED** [review]
The copy looks good to me 👍 

Functionality question: can they use a specific branch as a context also? 🤔

**Comment by sgrove** [comment]
@jasonbarry Didn't see anything that'd affect Netlify Graph token generation/injection negatively.

**Comment by jasonbarry** [comment]
> Functionality question: can they use a specific branch as a context also? 🤔

@codebyuma Yes -- because the toml supports `[context.<thing>.environment]`, where `<thing>` can be either a deploy context or the name of a branch, `netlify build --context <thing>` supports both. The calls to Envelope as of now don't, but that's what I'm working on next. The `branch:` prefix will also be supported, so e.g. `netlify build --context staging` and `netlify build --context branch:staging` will do the same thing (aka it'll be a synonym/alias). Do you think we'd need to describe both in the docs? Only touching on one or the other may be more straightforward, but I'll leave that up to you.

**Pull request merged**